infrastructure/src/main/java/org/corfudb/infrastructure/orchestrator/Orchestrator.java

```diff
@@ -75,6 +81,30 @@ void handleQuery(CorfuPayloadMsg<OrchestratorRequest> msg, ChannelHandlerContext
                 .payloadMsg(new OrchestratorResponse(resp)));
     }
 
+    void removeNode(CorfuPayloadMsg<OrchestratorRequest> msg, ChannelHandlerContext ctx, IServerRouter r) {
+        RemoveNodeRequest req = (RemoveNodeRequest) msg.getPayload().getRequest();
+
+        if (findUUID(req.getEndpoint()) != null) {
+            // An add node workflow is already executing for this endpoint, return
```

Comment:
Is this comment out of place or is the findUUID getting add node workflows in remove node ?

Reply:
Also this looks like similar code to what you do in addNode. Can we consolidate and have less code ?

Reply:
I refactored in the HOG branch that depends on this, but I'll push the change here.

Reply:
Removed, now we only have dispatch.  Its refactored.